### PR TITLE
working - needs tidy up and tests

### PR DESCRIPTION
**Pull request by zimboflyman**
This PR covers both EAR-790 and 791

### What is the context of this PR?

Validate missing values in Date type answer
Make sure that earliest and latest dates have X number of days before/after and not null

### How to review 

**Design**
https://overflow.io/s/AUOX1BTN?node=efa2355b&prototype

**Acceptance criteria (user perspective)**
Given I've created a validation rule for a date answer type
And left the value box blank
When I'm on the question page
Then the appropriate property should be highlighted and have an associated error message

Given I've created a validation rule for a date answer type
And left the value box blank
When I'm on the question page
Then the design tab should have an associated validation error

Given I've created a validation rule for a date answer type
And left the value box blank
When I'm on the question page
Then the left hand navigation should have an associated validation error on the specific question

### CODE REVIEW

**Review by MattElcock — CHANGES_REQUESTED** [review]
Hi Shane. Code looks good! Nice one. Just two refactoring things. 

Also, when you have some time (I'll hit you up on Slack too) could you explain the `validValidations` block you've added to the answer validations to me, please? I don't doubt it's necessary but I'm curious to know why.

File: eq-author/src/App/page/Design/Validation/AnswerValidation.js [hunk 1]
```diff
@@ -210,32 +237,56 @@ class AnswerValidation extends React.PureComponent {
...
+
+        const noValError = find(validationErrors.other, error =>
+          error.errorCode.includes("ERR_NO_VALUE")
+        );
+
+        const maxError = find(validationErrors.other, error =>
+          error.errorCode.includes("ERR_EARLIEST_AFTER_LATEST") || error.errorCode.includes("ERR_MIN_LARGER_THAN_MAX")
+        );
+
+        if (noValError || (noValError && maxError)) {
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author/src/App/page/Design/Validation/AnswerValidation.js]: aahhh - silly me -

File: eq-author/src/App/page/Design/Validation/AnswerValidation.js [hunk 2]
```diff
@@ -303,36 +354,11 @@ class AnswerValidation extends React.PureComponent {
...
-          validationMessage = ERR_NO_VALUE
-        } else {
-          validationMessage = MAX_GREATER_THAN_MIN
-        }
-      }
-
-      const propertyError = this.renderPropertyError(
-        validationErrors.other.length > 0,
-        validationMessage
+        // validationErrors
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author/src/App/page/Design/Validation/AnswerValidation.js]: wooops!

**Review by tbaggaley — APPROVED** [review]
Tested and working, looks good; Matt E's change suggestions incorporated. Will submit my refactoring changes as a PR after this goes through

**Review by jasehumphr — APPROVED** [review]
Looks Sweet

**Pull request merged**